server/service/apple_mdm.go

```diff
@@ -3438,6 +3471,249 @@ func ReconcileAppleProfiles(
 	return nil
 }
 
+func preprocessProfileContents(
+	ctx context.Context,
+	appConfig *fleet.AppConfig,
+	ds fleet.Datastore,
+	targets map[string]*cmdTarget,
+	profileContents map[string]mobileconfig.Mobileconfig,
+) error {
+
+	// This method replaces Fleet variables ($FLEET_VAR_<NAME>) in the profile contents, generating a unique profile for each host.
```

Comment:
to document: Apple profiles are signed using a pkcs7 envelope. If the IT admin uploads a signed profile with a Fleet variable, we'll replace the contents of the variable and invalidate the signature.

Reply:
Ok, made a note.